backend/sirius-web-persistence/src/main/java/org/eclipse/sirius/web/persistence/entities/CustomImageEntity.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.persistence.entities;
+
+import java.text.MessageFormat;
+import java.util.UUID;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+/**
+ * Represents a custom (user-supplied) image that can be used on representations.
+ *
+ * @author pcdavid
+ */
+@Entity
+@Table(name = "CustomImage")
+public class CustomImageEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private UUID id;
+
+    private String label;
+
+    @Column(name = "filename")
+    private String fileName;
```

Comment:
Why do we have both a filename and an id? Why is the filename needed?

- Since we have a label I imagine that the label will be displayed to the end user, not the filename
- Since the identify of the entity is defined by its id, why the filename?

Is it used only to check that we didn't contribute the same image twice for hardcoded images? For those images, the id could be derived from the filename.

Reply:
I hesitated about this. Having the filename was mostly to have "cleaner" URLs when accessing the images from `ImagesController`, but we probably don't care.
One issue though is that currently the filename is used to identify the file type and this the content-type (via the extension).
If we only keep the id, label and content, direct URLs would by of the form `/api/images/custom/${someUUID}` how would `ImageController` determine the content-type?

Or we can add a `content-type` column directly in the table and set it on import, but it feels like adding a subset of the information we already have with the filename (through the extension).

Or we can decide to only support SVGs, but that seems wrong.